src/box/replication.cc

```diff
@@ -1126,8 +1160,11 @@ replicaset_connect(const struct uri_set *uris,
 	while (!replicaset_is_connected(&state, appliers, count,
 					connect_quorum)) {
 		double wait_start = ev_monotonic_now(loop());
-		if (fiber_cond_wait_timeout(&state.wakeup, timeout) != 0)
+		if (fiber_cond_wait_timeout(&state.wakeup, timeout) != 0) {
+			if (fiber_is_cancelled())
+				diag_raise();
```

Comment:
You could use `fiber_testcancel` helper.

Reply:
Done.

```diff
--- a/src/box/replication.cc
+++ b/src/box/replication.cc
@@ -1170,8 +1170,7 @@ replicaset_connect(const struct uri_set *uris,
                                        connect_quorum)) {
                double wait_start = ev_monotonic_now(loop());
                if (fiber_cond_wait_timeout(&state.wakeup, timeout) != 0) {
-                       if (fiber_is_cancelled())
-                               diag_raise();
+                       fiber_testcancel();
                        break;
                }
                timeout -= ev_monotonic_now(loop()) - wait_start;
```